bin/src/hickory-dns.rs

```diff
@@ -453,10 +453,6 @@ fn run() -> Result<(), String> {
     if listen_addrs.is_empty() {
         listen_addrs.push(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)));
     }
-    let sockaddrs: Vec<SocketAddr> = listen_addrs
```

Comment:
Would prefer to split this change from the change that introduces the socket helpers.